src/index.js

```diff
@@ -0,0 +1,327 @@
+        console.log("DOWN " + cursorPosition);
+        if (textAreaRows.length >= 2) {
+            changeTextAreaCursorPosition(cursorPosition, cursorPosition);
+        }
+    }
+}
+
+function upButtonAction(event) {
+    if (event.type === 'mousedown' || event.type === 'keydown') {
+        const cursorPosition = calculateTextAreaCursorPositionAfterUp();
+        console.log('After UP position ' + cursorPosition);
+        changeTextAreaCursorPosition(cursorPosition, cursorPosition);
+    }
+}
+
+function commonShiftHandler(event) {
+    if (event.type === 'keydown' || event.type === 'mousedown'){
+        shiftDownButtonAction();
+    }
+    else if (event.type === 'keyup' || event.type === 'mouseup'){
+        shiftUpButtonAction();
+    }
+}
+
+function shiftUpButtonAction() {
+    shiftBtn = false;
+    if (capsBtn) {
+        capsOn();
+    }
+    else {
+        defaultOn();
+    }
+}
+
+function shiftDownButtonAction() {
+    shiftBtn = true;
+    if (altBtn){
+        keyBoardLanguageSwitcher();
+    }
+    shiftOn();
+}
+
+function commonAltHandler(event){
+    if (event.type === 'keydown' || event.type === 'mousedown'){
+        altDownButtonHandler();
+    }
+    else if (event.type === 'keyup' || event.type === 'mouseup'){
+        altUpButtonHandler();
+    }
+}
+
+function altUpButtonHandler() {
+    altBtn = false;
+}
+
+function altDownButtonHandler(){
+    altBtn = true;
+    if (shiftBtn){
+        keyBoardLanguageSwitcher();
+    }
+}
+
+function enterButtonAction(event) {
+    if (event.type === 'mousedown' || event.type === 'keydown')
+    {
+        const enter_value = '\n';
+        insertValueIntoTextArea(enter_value);
+    }
+}
+
+function capsLockButtonClickAction(keycode) {
+    if (!capsBtn) {
+        capsBtn = true;
+        setSelectedButton(keycode);
+        if (shiftBtn) {
+            shiftOn();
+        }
+        else {
+            capsOn();
+        }
+    }
+    else {
+        capsBtn = false;
+        defaultOn();
+        removeSelectedButton(keycode);
+    }
+}
+
+function capsOn(){
+    document.querySelectorAll("span.caps").forEach(elem => elem.classList.remove('hidden'));
+    document.querySelectorAll('span.default').forEach(elem => elem.classList.add('hidden'));
+    document.querySelectorAll('span.caseUp').forEach(elem => elem.classList.add('hidden'));
+}
+
+function shiftOn(){
+    document.querySelectorAll('span.default').forEach(elem => elem.classList.add('hidden'));
+    document.querySelectorAll('span.caseUp').forEach(elem => elem.classList.remove('hidden'));
+    document.querySelectorAll('span.caps').forEach(elem => elem.classList.add('hidden'));
+}
+
+function defaultOn(){
+    document.querySelectorAll("span.caps").forEach(elem => elem.classList.add('hidden'));
+    document.querySelectorAll('span.default').forEach(elem => elem.classList.remove('hidden'));
+    document.querySelectorAll('span.caseUp').forEach(elem => elem.classList.add('hidden'));
+}
```

Comment:
Combine these 3 functions to one. Add 'hidden' class to all and then remove it on parameter. (like in languages) You could even combine all this with language switcher, I suppose...